Swift/BRCrypto/BRCryptoWalletManager.swift

```diff
@@ -175,6 +193,25 @@ public final class WalletManager: Equatable {
         cryptoWalletManagerGive (core)
     }
 
+    static internal func defaultAddressScheme (currency: Currency) -> AddressScheme {
+        switch currency.code {
+        case Currency.codeAsBTC: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_BTC_SEGWIT)
+        case Currency.codeAsBCH: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_BTC_LEGACY)
+        case Currency.codeAsETH: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_ETH_DEFAULT)
+        default: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_GEN_DEFAULT)
+        }
+    }
+
+    static internal func defaultAddressSchemes (currency: Currency) -> [AddressScheme] {
```

Comment:
Also, what do we think about Network (instead of WalletManager) having defaultAddressScheme(s) (or defaultAddressScheme and supportedAddressSchemes), akin to supportedModes?

Reply:
We have `addressFor()` in Network so there is some nothing of address-related code there.

Reply:
If WalletManager has it it has to be a `static`; if Network has it is can be on the instance.

Reply:
Right, which I think is nice, no?

Reply:
Wait, a latter change has them on Network, I think.  Let me check.

Reply:
Sometime in the future they are on System:
```
   ///
    /// Check if `network` supports `scheme`
    ///
    /// - Parameters:
    ///   - network: the network
    ///   - scheme: the scheme
    ///
    /// - Returns: If supported `true`; otherwise `false`.
    ///
    public func supportsAddressScheme (network: Network, _ scheme: AddressScheme) -> Bool {
        return supportedAddressSchemes(network: network).contains (scheme)
    }

    ///
    /// Return the default AddressScheme for `network`
    ///
    /// - Parameter network: the network
    ///
    /// - Returns: The default AddressScheme
    ///
    public func defaultAddressScheme (network: Network) -> AddressScheme {
        switch network.currency.code {
        case Currency.codeAsBTC: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_BTC_SEGWIT)
        case Currency.codeAsBCH: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_BTC_LEGACY)
        case Currency.codeAsETH: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_ETH_DEFAULT)
        default: return AddressScheme (core: CRYPTO_ADDRESS_SCHEME_GEN_DEFAULT)
        }
    }

    ///
    /// Return the WalletManagerModes supported by `network`
    ///
    /// - Parameter network: the network
    ///
    /// - Returns: an aray of WalletManagerMode
    ///
    public func supportedModes (network: Network) -> [WalletManagerMode] {
        switch network.currency.code {
        case Currency.codeAsBTC: return [WalletManagerMode.api_only,
                                         WalletManagerMode.p2p_only]
        case Currency.codeAsBCH: return [WalletManagerMode.p2p_only]
        case Currency.codeAsETH: return [WalletManagerMode.api_only,
                                         WalletManagerMode.api_with_p2p_submit]
        default: return [WalletManagerMode.api_only]
        }
    }

    ///
    /// Check if `network` supports `mode`
    ///
    /// - Parameters:
    ///   - network: the network
    ///   - mode: the mode
    ///
    /// - Returns: If supported `true`; otherwise `false`
    ///
    public func supportsMode (network: Network, _ mode: WalletManagerMode) -> Bool {
        return supportedModes (network: network).contains (mode)
    }

    ///
    /// Return the default WalletManagerMode for `network`
    ///
    /// - Parameter network: the network
    ///
    /// - Returns: the default mode
    ///
    public func defaultMode (network: Network) -> WalletManagerMode {
        switch network.currency.code {
        case Currency.codeAsBTC: return WalletManagerMode.p2p_only
        case Currency.codeAsBCH: return WalletManagerMode.p2p_only
        case Currency.codeAsETH: return WalletManagerMode.api_only
        default: return WalletManagerMode.api_only
        }
    }
```

Reply:
Somehow they feel 'System-y'.  No?

Reply:
Interesting. I thought the modes were OK where they were and would have been okay with address schemes on the network too.

Reply:
`System` is your least favorite place for them?

Reply:
System is fine. I just don't see the value over network, which is the dictating factor for those calculations. If we think the System could, in the future, factor into that decision making, then System is the clear choice or even if we just want to hedge. Just my $0.02, I'm fine either way.